blog: newsletter november 2021 (#990)
tilt-dev/tilt.build: blog/_posts/2021-11-29-tilt-news-november-2021.md

---
slug: "tilt-news-november-2021"
date: 2021-11-29
author: lian
layout: blog
title: "Tilt News, November 2021"
image: "/assets/images/tilt-news-november-2021/pexels-frans-van-heerden-624015.jpg"
image_caption: "Aurora Borealis behind an igloo. Photo by Frans Van Heerden from <a href='https://www.pexels.com/photo/aurora-borealis-624015/'>Pexels</a>"
description: "introducing unsung heroes and another How-To video! 🎉"
youtubeId: Wur9HPjZ2Jo
tags:
  - news
---
Bonjour Tilteurs et Tilteuses,

> December is just around the corner, which means it’s time for the end of the year madness again. Hard to believe 2021 is almost over, isn’t it? We’re planning a cool 2021 recap for the end of the year, so there’ll be no December newsletter 😢  

Reviewer: ```suggestion
December is just around the corner, which means it’s time for the end of the year madness again. Hard to believe 2021 is almost over, isn’t it? We’re planning a cool 2021 recap for the end of the year, so we'll have that instead of a December newsletter!  
```

Please don't make me sad right in the beginning :)

But for now, the November edition comes packed with great reads, awesome features, and one or two glimpses behind the scenes!

## Tilt Mentions
We love to see more and more folks mentioning and discussing Tilt publicly! 

* Clear Street’s Ottavio Hartman wrote a blog post on [leveraging Tilt for their local dev environment](https://medium.com/@ottopo/523045dd2ac8) dubbed “localkube”.

> * In “[Defining the Developer Experience](https://dzone.com/articles/developer-experience)”, Kevin Taylor, VP of Engineering at Caylent, ponders the implications of Cloud Native development on the evolution of DevX. Tilt is mentioned together with Skaffold as tools to help develop on Kubernetes.

Reviewer: > Caylent

it's impossible not to read this as caliente lol

> * In his book [Effortless Cloud-Native App Development Using Skaffold](https://books.google.com/books?id=vNJCEAAAQBAJ&pg=PA233&lpg=PA233&dq=effortless+cloud-native+app+development+using+skaffold+tilt&source=bl&ots=NUwNtLM0Ci&sig=ACfU3U38wdnpWJF78uxTiNKkbgYDSnHAhQ&hl=en&sa=X&ved=2ahUKEwjx1Z3jnYn0AhXPSt8KHVsCBnMQ6AF6BAgTEAM#v=onepage&q=effortless%20cloud-native%20app%20development%20using%20skaffold%20tilt&f=false), Ashish Choudhary introduces Tilt as a possible alternative to Skaffold, 

Reviewer: ```suggestion
* In his book [Effortless Cloud-Native App Development Using Skaffold](https://books.google.com/books?id=vNJCEAAAQBAJ&pg=PA233&lpg=PA233&dq=effortless+cloud-native+app+development+using+skaffold+tilt&source=bl&ots=NUwNtLM0Ci&sig=ACfU3U38wdnpWJF78uxTiNKkbgYDSnHAhQ&hl=en&sa=X&ved=2ahUKEwjx1Z3jnYn0AhXPSt8KHVsCBnMQ6AF6BAgTEAM#v=onepage&q=effortless%20cloud-native%20app%20development%20using%20skaffold%20tilt&f=false), Ashish Choudhary introduces Tilt as a potential alternative to Skaffold, 
```

> Hopefully one day we’ll be able to add *your* take on Tilt too. Whether your experience with Tilt was superb or awful, we love to read about it! 🤓

Reviewer: ```suggestion
Hopefully one day we’ll be able to add *your* take on Tilt too. Whether your experience with Tilt was superb or not quite, we'd love to read about it! 🤓
```

## Tilt News
### Docs
Long-time Tilters have probably already noticed that we have overhauled our [docs](http://docs.tilt.dev) and introduced a [new, shiny tutorial](https://docs.tilt.dev/tutorial/index.html).  

> So far, our docs were mainly a collection of information by maintainers. From now on, we are designing our docs with our users in mind first. Rather than giving you all the information you could possibly need (and more), we’re trying to understand which problems you are trying to solve and how to best guide you through the solution finding process.  

Reviewer: ```suggestion
So far, our docs were mainly a collection of information by maintainers. From now on, we are designing our docs with our users journeys first and foremost. Rather than giving you all the information you could possibly need, we’re trying to understand which problems you are trying to solve and how to best guide you through the solution finding process.  
```

To read more about the changes, [visit our blog](https://blog.tilt.dev/2021/11/08/new-tutorial.html).

### Custom Builds
Often people ask us for help to optimize their docker builds. Unfortunately, some issues lie in the nature of the beast. However, did you know there are plenty of alternatives out there that work with Tilt?  
Check out [Nick’s blog post](https://blog.tilt.dev/2021/11/12/docker-build-alternatives.html) to learn how to use `kubectl build`, `ko`, and `pack` with Tilt!

### How-To Button 2: The Rebuttoning

> There are sequels nobody wants, but everyone knows are coming, like the tenth installment of “The Fast and the Furious”. Then there are sequels no one expected, but once they’re out you can’t imagine the world without them, like the exceptional [Terminator 2: Judgement Day](https://www.youtube.com/watch?v=nxr2SV5znwI).

Reviewer: ```suggestion
There are sequels nobody wants, but everyone knows are coming, like the tenth installment of “The Fast and the Furious.” Then there are sequels no one expected, but once they’re out you can’t imagine the world without them, like the exceptional [Terminator 2: Judgement Day](https://www.youtube.com/watch?v=nxr2SV5znwI).
```

Author: American English is weird

Reviewer: if you're ever in the mood for pain this is *chef's kiss* https://www.chicagomanualofstyle.org/home.html

Our second video in the How-To series is neither of those. It’s just a way to quickly learn how to combine Tilt UI buttons with other features, like passing along text input. Enjoy!

{% include youtubePlayer.html id=page.youtubeId %}

### Unsung heroes: our *favourite* feature

> There are Tilt features, that are quite obvious, like the possibility to [group resources visually](https://blog.tilt.dev/2021/08/09/resource-grouping.html) by label. But today we’ll talk about a feature that is frequently overlooked. Heck - some people don’t even know about it!  

Reviewer: ```suggestion
There are Tilt features that are quite obvious, like the possibility to [group resources visually](https://blog.tilt.dev/2021/08/09/resource-grouping.html) by label. But today we’ll talk about a feature that is frequently overlooked. Heck - some people don’t even know about it!  
```
